crate_universe/src/metadata/workspace_discoverer.rs

```diff
@@ -135,8 +135,13 @@ fn discover_workspaces_with_cache(
                                 explicit_workspace_path.display()
                             )
                         })?;
-                    actual_workspace_path =
-                        child_path.parent().unwrap().join(explicit_workspace_path);
+                    actual_workspace_path = child_path
+                        .parent()
+                        .unwrap()
+                        .join(explicit_workspace_path)
+                        .canonicalize_utf8()
```

Comment:
Is it possible to avoid canonicalizing the path? I don't think we want to resolve symlinks here, we just want a normalized path, right?

Reply:
I'm not entirely sure but I think you're correct. Since all we want in the end is to see whether `actual_workspace_path` is in `discovered_workspaces` then normalization should be sufficient and preferable.

Now, I don't think the standard library has such a function, neither does `camino`. I found some discussion about the feature in https://github.com/rust-lang/rfcs/issues/2208 but the issue is still open. Despite its promising name, `normpath` only performs canonicalization as well.

So, would adding a dependency on [`path-clean`](https://github.com/danreeves/path-clean) be ok? Or maybe you can point me to such a utility function in this codebase?

Reply:
> we just want a normalized path,

done in follow-up commit